src/Widgets/AppButton.vala

```diff
@@ -201,4 +200,26 @@ public class Slingshot.Widgets.AppButton : Gtk.Button {
         }
     }
 #endif
+
+    private bool create_context_menu (Gdk.Event e) {
+        menu = new Slingshot.AppContextMenu (desktop_id, desktop_path);
+        menu.app_launched.connect (() => {
+            app_launched ();
+        });
+
+        if (menu.get_children () != null) {
+            switch (e.type) {
+                case Gdk.EventType.KEY_PRESS:
+                    menu.popup_at_widget (this, Gdk.Gravity.EAST, Gdk.Gravity.CENTER, e);
+                    return true;
+                case Gdk.EventType.BUTTON_PRESS:
+                    menu.popup_at_pointer (e);
+                    return true;
+                default:
+                    break;
+            }
+        }
+
+        return false;
```

Comment:
Can we use `Gdk.EVENT_STOP` and `Gdk.EVENT_PROPAGATE` here to keep the code more glance-able?